mirrord/kube/src/api/runtime.rs

```diff
@@ -126,6 +129,8 @@ impl RuntimeData {
             .and_then(|status| status.container_statuses.as_ref())
             .ok_or_else(|| KubeApiError::missing_field(pod, ".status.containerStatuses"))?;
 
+        let guessed_container = container_name.is_none() && container_statuses.len() > 1;
```

Comment:
I'd consider issuing a warning only if we have multiple containers that are not on the default skip list. Otherwise we'd be warning pretty much always, if the cluster uses proxies :thinking:

Reply:
Fixed. Let me know if you think the iterator code is now too ugly and loops would be better.